parallaxinc/solo
# Fix array list population on create

Fixes issue in last attempt on #108 
All array blocks (except array_init) share handlers, so only one type needs to be tested fully, then just check the others for correct rendering.

**Test 1 - array name availability and tracking:**
1. Start with a blank workspace.
2. Drag out an array_fill block
- A waring should appear, stating that array "list" has not been initialized.
3. Draw out an array_init block.
- The warning should disappear.
4. Change the name of the variable.
- The array_fill block's name should change with it.
5. Drag out a second array init block.  Give it a name besides "list"
 - Check the menu of the array_fill block; it should contain both array name options
6. Save, close, re-open the file.
- Make sure all blocks have retained their names.
7. Delete all blocks except the 2 array_init blocks - make sure their names are something other than "list"
8. Drag out a new array_fill block.
- It should show a warning, but the other two array name options should be available

**Test 2 - element range warnings**
1. Start with a blank canvas
2. Drag out an array_init block
3. Drag out an array_get block
4. Type in a value above 10 in the element value input
- The block should throw a warning.
5. Save, close, reopen.
- Make sure the block values persist and don't revert to default values, and that the warnings persist as well.
5. Replace the number block with a variable block.
- The warning should disappear

**Test 3 - Rendering**
1. Drag out each type of array block
2. Change the name in the array_init block
- All of the blocks should track the changes to the name
3. Make changes to most/all of the numbers, values, etc.
4. Save, close, reopen
- Make sure all values retain their values and don't revert to defaults.
